airbyte-integrations/connectors/destination-aws-datalake/destination_aws_datalake/config_reader.py

```diff
@@ -5,10 +5,82 @@
 import enum
 
 
-class AuthMode(enum.Enum):
+class CredentialsType(enum.Enum):
     IAM_ROLE = "IAM Role"
     IAM_USER = "IAM User"
 
+    @staticmethod
+    def from_string(s: str):
+        if s == "IAM Role":
+            return CredentialsType.IAM_ROLE
+        elif s == "IAM User":
+            return CredentialsType.IAM_USER
+        else:
+            raise ValueError(f"Unknown auth mode: {s}")
+
+
+class OutputFormat(enum.Enum):
+    PARQUET = "Parquet"
+    JSONL = "JSONL"
+
+    @staticmethod
+    def from_string(s: str):
+        if s == "Parquet":
+            return OutputFormat.PARQUET
+        elif s == "JSONL":
+            return OutputFormat.JSONL
+        else:
+            raise ValueError(f"Unknown output format: {s}")
+
+
+class CompressionCodec(enum.Enum):
+    SNAPPY = "SNAPPY"
+    GZIP = "GZIP"
+    ZSTD = "ZSTD"
+    UNCOMPRESSED = "UNCOMPRESSED"
+
+    @staticmethod
+    def from_config(str: str):
+        if str == "SNAPPY":
+            return CompressionCodec.SNAPPY
+        elif str == "GZIP":
+            return CompressionCodec.GZIP
+        elif str == "ZSTD":
+            return CompressionCodec.ZSTD
+        elif str == "UNCOMPRESSED":
+            return CompressionCodec.UNCOMPRESSED
+        else:
+            raise ValueError(f"Unknown compression codec: {str}")
+
+
+class PartitionOptions(enum.Enum):
+    NONE = "NO PARTITIONING"
+    DATE = "DATE"
+    YEAR = "YEAR"
+    MONTH = "MONTH"
+    DAY = "DAY"
+    YEAR_MONTH = "YEAR/MONTH"
+    YEAR_MONTH_DAY = "YEAR/MONTH/DAY"
+
+    @staticmethod
+    def from_string(s: str):
+        if s == "NO PARTITIONING":
+            return PartitionOptions.NONE
+        elif s == "DATE":
+            return PartitionOptions.DATE
+        elif s == "YEAR":
+            return PartitionOptions.YEAR
+        elif s == "MONTH":
+            return PartitionOptions.MONTH
+        elif s == "DAY":
+            return PartitionOptions.DAY
+        elif s == "YEAR/MONTH":
+            return PartitionOptions.YEAR_MONTH
+        elif s == "YEAR/MONTH/DAY":
+            return PartitionOptions.YEAR_MONTH_DAY
+        else:
+            raise ValueError(f"Unknown partition option: {s}")
```

Comment:
since this is not a required config attribute, it would be better to fall back to the default option `PartitionOptions.NONE` in the `else` part in case `s` is empty. Right now, this change will make any existing connections fail.